libs/core/langchain_core/messages/utils.py

```diff
@@ -845,6 +862,775 @@ def list_token_counter(messages: Sequence[BaseMessage]) -> int:
         )
 
 
+def _runnable_generator(func: Callable) -> Callable:
+    @overload
+    def wrapped(
+        messages: Literal[None] = None, **kwargs: Any
+    ) -> Runnable[
+        Union[MessageLikeRepresentation, Sequence[MessageLikeRepresentation]],
+        Union[BaseMessage, List[BaseMessage]],
+    ]: ...
+
+    @overload
+    def wrapped(
+        messages: Sequence[Union[BaseMessage, Dict, Tuple]], **kwargs: Any
+    ) -> List[BaseMessage]: ...
+
+    @overload
+    def wrapped(messages: MessageLikeRepresentation, **kwargs: Any) -> BaseMessage: ...
+
+    def wrapped(
+        messages: Union[
+            MessageLikeRepresentation, Sequence[MessageLikeRepresentation], None
+        ] = None,
+        **kwargs: Any,
+    ) -> Union[
+        BaseMessage,
+        List[BaseMessage],
+        Runnable[
+            Union[MessageLikeRepresentation, Sequence[MessageLikeRepresentation]],
+            Union[BaseMessage, List[BaseMessage]],
+        ],
+    ]:
+        from langchain_core.runnables.base import RunnableGenerator
+
+        if messages is not None:
+            return func(messages, **kwargs)
+        else:
+
+            def transform(input_: Iterator, **kwargs: Any) -> Iterator:
+                block_indexes = set()
+                for x in input_:
+                    msg = func(x, **kwargs)
+                    # Special handling for transforming an OpenAI stream to an
+                    # Anthropic stream.
+                    if isinstance(msg, AIMessageChunk) and isinstance(
+                        msg.content, list
+                    ):
+                        tool_use_ct = 0
+                        for block in msg.content:
+                            if not isinstance(block, dict):
+                                continue
+                            if "index" in block:
+                                block_indexes.add(block["index"])
+                            elif block.get("type") == "tool_use":
+                                block["index"] = max(len(block_indexes), *block_indexes)
+                                msg.tool_call_chunks[tool_use_ct]["index"] = block[
+                                    "index"
+                                ]
+                            else:
+                                pass
+
+                            if block.get("type") == "tool_use":
+                                tool_use_ct += 1
+                    else:
+                        block_indexes.add(0)
+                    yield msg
+
+            return RunnableGenerator(partial(transform, **kwargs), name=func.__name__)
+
+    wrapped.__doc__ = func.__doc__
+    return wrapped
+
+
+@_runnable_generator
+def format_messages_as(
+    messages: Union[MessageLikeRepresentation, Sequence[MessageLikeRepresentation]],
+    *,
+    format: Literal["openai", "anthropic"],
+    text_format: Literal["string", "block"],
```

Comment:
```suggestion
    format: Literal["openai", "anthropic", "openai_text", "anthropic_text"],
```
might be confusing to have `text_format` option if we ever introduce something else that doesn't offer one or the other

Reply:
something that doesn't offer string or block options?

Reply:
yep - for example if new_model_provider_c only supports blocks and not string content in their api for some reason. Or anthropic introduces a block_2 format. Keeping it 1-dimensional is easier to maintain in that case.